ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala

```diff
@@ -367,11 +367,10 @@ object RecoveringIndexerSpec {
             }
           })
         })(handle => {
-          val complete = handle.stop()
-          complete.onComplete { _ =>
+          handle.stop().transform { complete =>
```

Comment:
What is the difference? (Except for the nicer construction)

Reply:
I think previously we did:
```
val complete = handle.stop()
complete.onComplete { doStuff() }
complete // return complete immediately without waiting for doStuff()
```

and now it is:
```
handle.stop().transform { complete =>
doStuff() // do stuff when stop() completes
complete // return complete but this time only when doStuff() is completed
}

Reply:
I see, thanks

Reply:
I think: Before, the `.onComplete` handler and the `.map` transformation could run in any order. With this change, it is guaranteed that the subscription is removed first.

Reply:
In the past the lambda in onComplete was racing against the client code attached to the returned future. Now the lambda is executed strictly before.